Warnings with clang 12

There are compilation warnings with 

 - OS version: MacOs 10.15
 - Compiler version: clang version 12.0.0 
 - Package version: HEAD
 
```
In file included from /usr/local/ilcsoft/sio/v00-00-03/source/include/sio/api.h:360:
/usr/local/ilcsoft/sio/v00-00-03/source/include/sio/io_device.h:164:5: warning: explicitly defaulted copy constructor is implicitly deleted [-Wdefaulted-function-deleted]
    write_device( const write_device & ) = default ;
    ^
/usr/local/ilcsoft/sio/v00-00-03/source/include/sio/io_device.h:268:25: note: copy constructor of 'write_device' is implicitly deleted because field '_buffer' has a deleted copy constructor
    buffer              _buffer ;
                        ^
/usr/local/ilcsoft/sio/v00-00-03/source/include/sio/buffer.h:223:5: note: 'buffer' has been explicitly marked deleted here
    buffer( const buffer& ) = delete ;
    ^
In file included from /usr/local/ilcsoft/sio/v00-00-03/source/main/sio-dump.cc:4:
In file included from /usr/local/ilcsoft/sio/v00-00-03/source/include/sio/api.h:360:
/usr/local/ilcsoft/sio/v00-00-03/source/include/sio/io_device.h:168:19: warning: explicitly defaulted copy assignment operator is implicitly deleted [-Wdefaulted-function-deleted]
    write_device& operator=( const write_device & ) = default ;
                  ^
/usr/local/ilcsoft/sio/v00-00-03/source/include/sio/io_device.h:268:25: note: copy assignment operator of 'write_device' is implicitly deleted because field '_buffer' has a deleted copy assignment operator
    buffer              _buffer ;
                        ^
/usr/local/ilcsoft/sio/v00-00-03/source/include/sio/buffer.h:225:13: note: 'operator=' has been explicitly marked deleted here
    buffer& operator=( const buffer& ) = delete ;
            ^
In file included from /usr/local/ilcsoft/sio/v00-00-03/source/main/sio-dump.cc:4:
/usr/local/ilcsoft/sio/v00-00-03/source/include/sio/api.h:472:17: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
      rec_buf = std::move( device.take_buffer() ) ;
                ^
/usr/local/ilcsoft/sio/v00-00-03/source/include/sio/api.h:472:17: note: remove std::move call here
      rec_buf = std::move( device.take_buffer() ) ;
                ^~~~~~~~~~~                     ~~
3 warnings generated.
```

--- a/source/include/sio/api.h
+++ b/source/include/sio/api.h
@@ -469,7 +469,7 @@ namespace sio {
       device.data( rec_info._options ) ;
       device.data( rec_info._data_length ) ;
       // get back the buffer
-      rec_buf = std::move( device.take_buffer() ) ;
+      rec_buf = device.take_buffer() ;
     }
     catch( sio::exception &e ) {
       SIO_RETHROW( e, sio::error_code::io_failure, "Couldn't compress record buffer" ) ;

--- a/source/include/sio/io_device.h
+++ b/source/include/sio/io_device.h
@@ -161,11 +161,11 @@ namespace sio {
     /// No default constructor
     write_device() = delete ;
     /// Default copy constructor
-    write_device( const write_device & ) = default ;
+    write_device( const write_device & ) = delete ;
     /// Default move constructor
     write_device( write_device && ) = default ;
     /// Default assignement operator
-    write_device& operator=( const write_device & ) = default ;
+    write_device& operator=( const write_device & ) = delete ;
     /// Default move assignement operator
     write_device& operator=( write_device && ) = default ;
     /// Default destructor

